src/applications/bmqstoragetool/m_bmqstoragetool_journalfileprocessor.cpp

```diff
@@ -36,14 +38,49 @@
 namespace BloombergLP {
 namespace m_bmqstoragetool {
 
+namespace {
+template <typename T>
+T getValue(const mqbs::JournalFileIterator*  jit,
+           const Parameters::SearchValueType valueType);
+
+template <>
+bsls::Types::Uint64 getValue(const mqbs::JournalFileIterator*  jit,
+                             const Parameters::SearchValueType valueType)
+{
+    // PRECONDITIONS
+    BSLS_ASSERT(jit);
+    BSLS_ASSERT(valueType == Parameters::e_TIMESTAMP ||
+                valueType == Parameters::e_OFFSET);
+
+    return (valueType == Parameters::e_TIMESTAMP)
+               ? jit->recordHeader().timestamp()
+               : jit->recordOffset();
+}
+
+template <>
+CompositeSequenceNumber getValue(const mqbs::JournalFileIterator*  jit,
+                                 const Parameters::SearchValueType valueType)
+{
+    // PRECONDITIONS
+    BSLS_ASSERT(jit);
+    BSLS_ASSERT(valueType == Parameters::e_SEQUENCE_NUM);
+
+    return CompositeSequenceNumber(jit->recordHeader().primaryLeaseId(),
+                                   jit->recordHeader().sequenceNumber());
+}
```

Comment:
I'd better make a comparator function and pass it to `moveToLowerBound`. It can significantly simplify the logic:

```
bool comparatorFunction(const mqbs::JournalFileIterator&  jit,
                                         const Parameters& parameters)
{
  if (valueType == Parameters::e_TIMESTAMP) {
    return jit.recordHeader().timestamp() > parameters.d_timestampGt;
  } else if () { ... } esle { ... }
}
```

Reply:
Passed functor class `LessThanLowerBoundFn` for comparison and got rid of template methods.